src/value/loader.rs

```diff
@@ -0,0 +1,462 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+use crate::result::IonResult;
+use crate::value::owned::{OwnedElement, OwnedSequence, OwnedStruct, OwnedSymbolToken, OwnedValue};
+use crate::value::AnyInt;
+use crate::IonType;
+use ion_c_sys::reader::{IonCReader, IonCReaderHandle};
+use ion_c_sys::ION_TYPE;
+use std::convert::{TryFrom, TryInto};
+
+use super::owned::text_token;
+
+/// TODO add/refactor trait/implementation for borrowing over some context
+///      we could make it generic with generic associated types or just have a lifetime
+///      scoped implementation
+
+/// Loads Ion data into [`Element`] instances.
```

Comment:
The docs around this trait are written in terms of the generics we plan to have eventually. For the time being, should we just refer to `OwnedElement` everywhere to align with the current implementation?

Reply:
I'd like to move this to `BorrowedElement` sooner than later to make sure this is doable with this API.  Also, the real docs for these APIs are in the traits and not the implementations, so not clear that would help anyone.  That said if you think this would be clearer, I could do that, I really do want folks to think of it in terms of `Element`, in other words, the lack of *existential types* in more useful places is really the reason why `OwnedElement` pops out.